integrate/simpsons_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package integrate
+
+import (
+	"math"
+	"testing"
+
+	"gonum.org/v1/gonum/floats"
+)
+
+func TestSimpsons(t *testing.T) {
+	const N = 1e6
+	x := floats.Span(make([]float64, N), 0, 1)
+	for i, test := range []struct {
```

Comment:
I would include in the tests:
* test cases that have a small number of locations, say `3 <= len(x) <= 12`
* a constant function
* test-case specific tolerance. A global tolerance must accommodate the least accurate result which might hide bugs.

Reply:
- I added a few tests with small number ranges

I tried adding a tolerance value, but it looks like floats.EqualWithinAbs behaves differently when I pass the tolerance through a struct:
```
floats.EqualWithinAbs(v, test.want, 1e-12)
```
```
// test.tol := 1e-6
floats.EqualWithinAbs(v, test.want, test.tol)
```

Reply:
Must be a bug in your code.